### PR TITLE
Transformation Update

### PR DESCRIPTION
**Pull request by NicerNewerCar**
* Adds DICOM2AUT tfm
* Splits `{bone}.tfm` into `{bone}_t.tfm` and `{bone}_scale.tfm`
* Renames and addresses issues with Slicer2Autoscoper transform (now PVOL2AUT)

### CODE REVIEW

**Comment by amymmorton** [comment]
building now... and will confirm, but a comment on the commit message re: splitting the .tfm:
We do want all three variants:  bone.tf   boneScaleOnly.tfm and boneTransOnly.tfm

**Comment by NicerNewerCar** [comment]
> building now... and will confirm, but a comment on the commit message re: splitting the .tfm: We do want all three variants: bone.tf boneScaleOnly.tfm and boneTransOnly.tfm

Okay, I will update this now to include the {bone}.tfm again

**Comment by amymmorton** [comment]
- [ ] Pvol 2 aut  

 suggest

- [ ]  export of the AUT(bone).stl  (LPS)
- [ ]  change  bone_t  to inv(bone_t)
- [ ]  the DICOM2AUT tfm is combined transform of the image position to aut  AND RAS->LPS - suggest naming explicitly?  (if user segments in mimics and applies this tform- (staying LPS) to the bone model or a cs gen from the anatomy it will be erroneous.

**Comment by NicerNewerCar** [comment]
> * [ ]   change  bone_t  to inv(bone_t)

Should this also apply to the naming of the bone_scale (ie. inv(bone_scale))

> * [ ]   the DICOM2AUT tfm is combined transform of the image position to aut  AND RAS->LPS - suggest naming explicitly?  (if user segments in mimics and applies this tform- (staying LPS) to the bone model or a cs gen from the anatomy it will be erroneous.

Would something like `DICOM(RAS)2AUT(LPS)` work or were you thinking of something else?

File: AutoscoperM/AutoscoperM.py [hunk 1]
```diff
@@ -757,30 +767,29 @@ def onLoadPV(self):
             transformSubDir = self.ui.tfmSubDir.text
 
             vols = glob.glob(os.path.join(mainOutputDir, volumeSubDir, "*.tif"))
-            tfms = glob.glob(os.path.join(mainOutputDir, transformSubDir, "*.tfm"))
+            tfms_t = glob.glob(os.path.join(mainOutputDir, transformSubDir, "*_t.tfm"))
```
Comment (jcfr) [inline AutoscoperM/AutoscoperM.py:770]: Independently of this pull request, we should consider documentating the different transform files generated by Autoscoper/AutoscoperM.

May be by updating the following section: https://autoscoper.readthedocs.io/en/latest/file-specifications/index.html

File: AutoscoperM/AutoscoperM.py [hunk 2]
```diff
@@ -1467,54 +1524,64 @@ def applyOrigin2DicomTransform(
         return transform
 
     @staticmethod
-    def applySlicer2AutoscoperTransform(
+    def applyPVol2AutTransform(
         transform: vtk.vtkMatrix4x4,
-        slicer2AutoscoperNode: slicer.vtkMRMLLinearTransformNode,
-        applyAutoscoper2Slicer: bool = False,
+        PVol2AutNode: slicer.vtkMRMLLinearTransformNode,
```
Comment (jcfr) [inline AutoscoperM/AutoscoperM.py:1529]: `PVol2AutNode` -> `pVol2AutNode`

File: AutoscoperM/AutoscoperM.py [hunk 3]
```diff
@@ -1467,54 +1524,64 @@ def applyOrigin2DicomTransform(
...
         volumeNode.GetRASBounds(bounds)
         volSize = [abs(bounds[i + 1] - bounds[i]) for i in range(0, len(bounds), 2)]
 
-        slicer2autoscoper = vtk.vtkMatrix4x4()
-        slicer2autoscoper.Identity()
-        # Rotation matrix for a 180 y-axis rotation
-        slicer2autoscoper.SetElement(0, 0, -slicer2autoscoper.GetElement(0, 0))
-        slicer2autoscoper.SetElement(2, 2, -slicer2autoscoper.GetElement(2, 2))
-        slicer2autoscoper.SetElement(1, 3, -volSize[1])  # Offset -Y
+        pvol2aut = vtk.vtkMatrix4x4()
```
Comment (jcfr) [inline AutoscoperM/AutoscoperM.py:1562]: `pvol2aut` -> pVol2Aut

**Review by jcfr — COMMENTED** [review]
After addressing few nitpicks and confirming there are no regression, unless @amymmorton has additional comments, this is good for integration :rocket:

**Comment by amymmorton** [comment]
The bone_t.tfm are in RAS not LPS (name it explicitly?)
(ground truth data prev sent 7/16  vs  this branch)
image


- [ ] GT AUT mc1 stl and AUT stl generated in this branch are spatially aligned 
- [ ] mc1-DICOM2AUT.tfm transforms the segmented mc1 stl to AUT space in RAS -slicer

the output tra (with LPS-RAS) tform applied is correct

**Review by amymmorton — APPROVED** [review]

**Pull request merged**